#数据计topic
#bag_path: /home/udi/Downloads/dataset/shenlan/for_calib.bag     #数据集相对路径
#截取数据集的时间段：                     注意：此处最好不要超过20s的数据，否则会导致SO3样条轨迹不收敛
bag_start: 5      #起始时间
bag_end: 130       #结束时间
#bag_durr: 20      #时长,如果小于０就截取到数据集结束时间
key_frame_num: 15 #用于svd计算初始旋转外参的关键帧数

data_path: /slam_data   # 数据存放路径



#kitti数据集测量
#measurements:
#  lidar:
#    topic_name: /kitti/velo/pointcloud
#    frame_id: velo_link
#    queue_size: 100000
#    frequency: 10
#  imu:
#    topic_name: /kitti/oxts/imu
#    frame_id: imu_link
#    queue_size: 1000000
#  gnss:
#    topic_name: /kitti/oxts/gps/fix
#    frame_id: imu_link
#    queue_size: 1000000

#楼下数据集测量
#measurements:
#  lidar:
#    topic_name: /top/rslidar_points
#    frame_id: /rslidar
#    queue_size: 100000
#    frequency: 10
#  imu:
#    topic_name: /sys_data_hub/imu_data
#    frame_id: world
#    queue_size: 1000000
#  gnss:
#    topic_name: /novatel718d/pos
#    frame_id: gps
#    queue_size: 1000000


#  velocity:
#    topic_name: /novatel718d/vel
#    frame_id: gps
#    queue_size: 10000000

##紫荆山数据集测量
measurements:
  lidar:
    topic_name: /top/rslidar_points
    frame_id: /rslidar
    queue_size: 100000
    frequency: 10
#  imu:
#    topic_name: /udi/imu_s/data
#    frame_id: imu_link
#    queue_size: 1000000
  imu:
    topic_name: /rion/yaw_data
    frame_id: /rion/yaw
    queue_size: 100000
  gnss:
    topic_name: /novatel718d/pos
    frame_id: gps
    queue_size: 1000000

# 匹配
registration_method: NDT   # 选择点云匹配方法，目前支持：NDT
# 各配置选项对应参数
## 匹配相关参数
NDT:
  res : 1.0             #分辨率
  step_size : 0.1       #步长
  trans_eps : 0.01      #迭代停止条件：变换阈值
  max_iter : 100         #最大迭代次数
  num_thr: 4            #线程数
  neighbor_search_method: pclomp::DIRECT7        #邻近搜索办法

# 局部地图参数
key_frame_distance: 2.5 # 关键帧距离
key_frame_diff_angle: 5.0
local_frame_num: 20

#点云滤波参数
frame_filter: voxel_filter # 选择当前帧点云滤波方法，目前支持：voxel_filter、no_filter
local_map_filter: voxel_filter # 选择滑窗地图点云滤波方法，目前支持：voxel_filter、no_filter
global_map_filter: voxel_filter # 选择滑窗地图点云滤波方法，目前支持：voxel_filter、no_filter

# 局部地图
# 局部地图从全局地图切割得到，此处box_filter_size是切割区间
# 参数顺序是min_x, max_x, min_y, max_y, min_z, max_z
box_filter_size: [-150.0, 150.0, -150.0, 150.0, -150.0, 150.0]
#local_map_filter: voxel_filter # 选择滑窗地图点云滤波方法，目前支持：voxel_filter、no_filter


## 滤波相关参数
voxel_filter:
  local_map:
    leaf_size: [0.6, 0.6, 0.6]
  frame:
    leaf_size: [0.5, 0.5, 0.5]
  global_map:
    leaf_size: [0.6, 0.6, 0.6]